process/smartContract/hooks/blockChainHook.go

```diff
@@ -396,10 +402,17 @@ func (bh *BlockChainHookImpl) LastTimeStamp() uint64 {
 
 // LastTimeStampMs returns the timeStamp in milliseconds from the last committed block
 func (bh *BlockChainHookImpl) LastTimeStampMs() uint64 {
-	if !check.IfNil(bh.blockChain.GetCurrentBlockHeader()) {
-		return common.ConvertTimeStampSecToMs(bh.blockChain.GetCurrentBlockHeader().GetTimeStamp())
+	if check.IfNil(bh.blockChain.GetCurrentBlockHeader()) {
+		return 0
 	}
-	return 0
+
+	timestamp := bh.blockChain.GetCurrentBlockHeader().GetTimeStamp()
+	timestampMs := common.ConvertTimeStampSecToMs(timestamp)
+	if common.IsSupernovaRoundActivated(bh.enableEpochsHandler, bh.enableRoundsHandler) {
+		timestampMs = timestamp
+	}
```

Comment:
maybe worth considering a new common method, GetTimestamps which will return timestamp and timestampMs, in order to avoid reusing this pattern?

Reply:
refactor to use common function